extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/devconsole/HibernateOrmDevConsoleInfoSupplier.java

```diff
@@ -57,37 +81,60 @@ public static void clearData() {
         INSTANCE.persistenceUnits.clear();
     }
 
-    private static String generateDDL(SchemaExport.Action action, Metadata metadata, ServiceRegistry serviceRegistry,
+    private static String generateDDL(Action action, Metadata metadata, ServiceRegistry ssr,
             String importFiles) {
-        SchemaExport schemaExport = new SchemaExport();
-        schemaExport.setFormat(true);
-        schemaExport.setDelimiter(";");
-        schemaExport.setImportFiles(importFiles);
-        StringWriter writer = new StringWriter();
+        final HibernateSchemaManagementTool tool = (HibernateSchemaManagementTool) ssr.getService(SchemaManagementTool.class);
+        Map<String, Object> config = new HashMap<>(ssr.getService(ConfigurationService.class).getSettings());
+        config.put(AvailableSettings.HBM2DDL_DELIMITER, ";");
+        config.put(AvailableSettings.FORMAT_SQL, true);
+        config.put(AvailableSettings.HBM2DDL_IMPORT_FILES, importFiles);
+        ExceptionHandlerCollectingImpl exceptionHandler = new ExceptionHandlerCollectingImpl();
```

Comment:
I don't know what's the reasoning behind this change, but if we change this we'll need to update non-regression tests in Hibernate ORM, which still simply use `SchemaExport`:

https://github.com/hibernate/hibernate-orm/blob/7e629957e45e1c1466eb0b7f9cb9c316c4705fa3/hibernate-core/src/test/java/org/hibernate/orm/test/boot/database/qualfiedTableNaming/DefaultCatalogAndSchemaTest.java#L638-L666

Reply:
The reason for the change is that `SchemaExport` no longer exists in ORM core - it was moved to the ANT specific tooling jar.
Funny, your integration test still works as the testsuite is depending on that too...  I guess theoreritcally we could pull in that dependency in Quarkus too, but it seems unnecessary and I suspect we could get in trouble with native compilation reaching into missing symbols (e.g. ANT being referenced).

So the above code replicates the equivalent functionality, just slightly less convenient.

Thanks for pointing out the need to change the ORM test though, I'll create an HHH to track; this also suggests ORM now lacks a "nice to use" SPI for such changes since the nicer `SchemaExport` is no longer there - perhaps we could introduce that too, or discuss restoring the original position.

Reply:
ORM issue is https://hibernate.atlassian.net/browse/HHH-16207 , and I'll be adding a link to it in the comments as well.